src/TesterInternal/SerializerTests/CustomSerializerTests.cs

```diff
@@ -132,5 +133,13 @@ public void Serialize_GrainMethodAsyncReturnOnlyType()
         {
             Assert.IsNotNull(SerializationManager.GetSerializer(typeof(SerializerTestClass6)), "No serializer generated for return type of parameterless Task grain method");
         }
+        
+        [TestMethod, TestCategory("Functional"), TestCategory("Serialization")]
+        public void Serialize_AsyncObserverArgumentType()
+        {
+            var z = new ClosedTypeStreamObserver(TraceLogger.GetLogger("SerializationTests", TraceLogger.LoggerType.Runtime));
+
+            Assert.IsNotNull(SerializationManager.GetSerializer(typeof(AsyncObserverArg)), "No serializer generated for argument type of async observer");
```

Comment:
Maybe also worth trying to actually serialize it as well? make sure it does not throw?